### PR TITLE
PosRec minitree unblinding

### PR DESCRIPTION
**Pull request by pdeperio**
The ```PositionReconstruction``` minitrees (ROOT files on disk) are mistakenly blinded due to [this line](https://github.com/XENON1T/hax/blob/v2.3.3/hax/treemakers/posrec.py#L122). @feigaodm noticed after the recent ```s2 < 200``` unblinding attempt (https://github.com/XENON1T/hax/pull/208), all the ```PositionReconstruction``` variables of events in the previously blinded region were ```NaN```.

To not stop the analysis, I did something naughty and commented out the blinding in ```minitrees.py``` [here](https://github.com/XENON1T/hax/blob/v2.3.3/hax/minitrees.py#L404) in my local installation:
```
    #if (hax.unblinding.unblinding_selection not in preselection and
    #    ('Corrections' in treemakers or
    #     hax.treemakers.corrections.Corrections in treemakers) and
    #        hax.unblinding.is_blind(run_id)):
    #    preselection = [hax.unblinding.unblinding_selection] + preselection
```
and regenerated the minitrees with version ```1.2``` as of today.

Do not merge this yet, in case someone is using some non-tagged runs that haven't been regenerated by cax yet, and end up regenerating themselves with the blinded version. We need to figure out how to fix [that line](https://github.com/XENON1T/hax/blob/v2.3.3/hax/treemakers/posrec.py#L122) and include in this PR.

### CODE REVIEW

**Comment by feigaodm** [comment]
@pdeperio Is this still needed, I think it's not

**Comment by pdeperio** [comment]
It is. Still no solution to the problem: PosRec minitree needs to be regenerated if you want to unblind.

File: hax/minitrees.py [hunk 1]
```diff
@@ -398,11 +398,11 @@ def load_single_dataset(run_id, treemakers, preselection=None, force_reload=Fals
 
     # Apply the unblinding selection if required. Normally this is already done by minitrees.load, but perhaps someone calls
     # load_single_dataset_directly.
-    if (hax.unblinding.unblinding_selection not in preselection and
-        ('Corrections' in treemakers or
-         hax.treemakers.corrections.Corrections in treemakers) and
-            hax.unblinding.is_blind(run_id)):
-        preselection = [hax.unblinding.unblinding_selection] + preselection
+    #if (hax.unblinding.unblinding_selection not in preselection and
```
Comment (tunnell) [inline hax/minitrees.py]: __[Flake8]__

[E265] block comment should start with '# '

File: hax/minitrees.py [hunk 2]
```diff
@@ -354,7 +354,7 @@ def load_single_minitree(run_id,
     return skimmed_data
 
 
-def load_single_dataset(run_id, treemakers, preselection=None, force_reload=False, event_list=None):
+def load_single_dataset(run_id, treemakers, preselection=None, force_reload=False, event_list=None, bypass_blinding=False):
```
Comment (tunnell) [inline hax/minitrees.py]: __[Flake8]__

[E501] line too long (123 > 120 characters)

File: hax/minitrees.py [hunk 3]
```diff
@@ -370,6 +370,8 @@ def load_single_dataset(run_id, treemakers, preselection=None, force_reload=Fals
 
     :param event_list: List of event numbers to visit. Disables load from / save to file.
 
+    :bypass_blinding: Flag to disable blinding cut (WARNING: analysts should not use this, only for production! See #211)
```
Comment (tunnell) [inline hax/minitrees.py]: __[Flake8]__

[E501] line too long (121 > 120 characters)

**Comment by feigaodm** [comment]
finally a solution 👍 did you check it on blinded data somehow?

**Comment by pdeperio** [comment]
Just implemented @JelleAalbers's suggestion above (https://github.com/XENON1T/hax/pull/211#issuecomment-394107014).

Yes, tested locally on background run 20626.

This can now be re-reviewed and merged if you approve. (Then @ershockley should not need local installation anymore for production.)

**Pull request merged**